### PR TITLE
Externalize oc tag

### PR DESCRIPTION
**Pull request by soltysh**
/assign @juanvallejo

### CODE REVIEW

File: pkg/oc/cli/tag/tag.go [hunk 1]
```diff
@@ -417,10 +403,18 @@ func (o TagOptions) Run() error {
...
-				if _, ok := target.Spec.Tags[destTag]; !ok {
+				if _, ok := imageutil.SpecHasTag(target, destTag); !ok {
 					return fmt.Errorf("destination tag %s/%s does not exist.\n", o.destNamespace[i], destNameAndTag)
 				}
-				delete(target.Spec.Tags, destTag)
+				// delete tag
+				tags := []imagev1.TagReference{}
+				for i := range target.Spec.Tags {
+					t := target.Spec.Tags[i]
+					if t.Name != destTag {
```
Comment (juanvallejo) [inline pkg/oc/cli/tag/tag.go]: nit: might improve readability a bit if we `continue` on the inverse condition:

if t.Name == destTag {
  continue
}

tags = append(tags, t)

File: pkg/oc/cli/tag/tag.go [hunk 2]
```diff
@@ -529,17 +523,23 @@ func (o TagOptions) Run() error {
...
 					delete(target.Annotations, imageapi.DockerImageRepositoryCheckAnnotation)
 					istag.Tag.Generation = nil
 				}
 			}
-			target.Spec.Tags[destTag] = *istag.Tag
+			// update tag
+			for i := range target.Spec.Tags {
+				t := target.Spec.Tags[i]
+				if t.Name == destTag {
+					t = *istag.Tag
```
Comment (juanvallejo) [inline pkg/oc/cli/tag/tag.go]: we could probably break after we've found the tag to update

**Comment by juanvallejo** [comment]
few nits, but otherwise lgtm

/lgtm

File: pkg/oc/cli/tag/tag.go (hunk 1 above)
Comment (soltysh) [reply pkg/oc/cli/tag/tag.go]: I find the current approach more readable, so I'll leave it as is.

File: pkg/oc/cli/tag/tag.go (hunk 2 above)
Comment (soltysh) [reply pkg/oc/cli/tag/tag.go]: Good point.

**Comment by soltysh** [comment]
/hold
until I fix the error

**Comment by soltysh** [comment]
Fixed, tagging back.

**Pull request merged**